cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -626,4 +628,84 @@ protected void kafkaHeapOptions(List<EnvVar> envVars, double dynamicHeapFraction
             envVars.add(buildEnvVar(ENV_VAR_KAFKA_HEAP_OPTS, trim));
         }
     }
+
+    public static int getInteger(Map<String, String> data, String key, int defaultValue) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map -> will be used default value
+                return defaultValue;
+            }
+            return Integer.parseInt(data.get(key));
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted.";
+            if (data.get(key).length() == 0) {
+                msg = " is empty.";
+            }
+            throw new InvalidConfigMapException(msg, key);
+        }
+    }
+
+    public static Storage getStorage(Map<String, String> data, String key) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(" is corrupted.", key);
+        }
+    }
+
+    public static String getString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(" is empty.", key);
+        }
+        return data.get(key);
+    }
+
+    public static String getJSONcorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+                .replace("\"", "").replace(" ", "")
+                .replace("\n", "").split(",");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static String getConfig(Map<String, String> data, String key) {
+        String config = "";
+        try {
+            config = data.get(key);
+            Map<String, Object> map = new JsonObject(config).getMap();
```

Comment:
So that you know: The CO started off using the Vertx JSON parsing API, but we're now drifting in the direction of using the Jackson API directly. The Jackson API exposes a little more information in its exceptions, which might have made `getJSONcorruptionBlame()` easier to write and less fragile. Generally speaking it's not good practice to try to parse exception messages for the same reason that it's not great to assert on them: It will break should the format of the exception message change. In this case it's especially bad because the messages are coming from a dependency you don't control so any breakage is outside your control.

I think it's OK to leave this using the Vertx Json API for now, but it's something we will have to tidy up, so please could you open an issue for it.